tests/unit/test_snap.py

```diff
@@ -551,26 +551,36 @@ def raise_error(cmd, **kwargs):
     def test_snap_set(self, mock_subprocess):
         foo = snap.Snap("foo", snap.SnapState.Present, "stable", "1", "classic")
 
-        foo.set({"bar": "baz"})
+        configs = {
+            "true": True,
+            "false": False,
+            "null": None,
+            "integer": 1,
+            "float": 2.0,
+            "list": [1, 2.0, True, False],
+            "dict": {
+                "true": True,
+                "false": False,
+                "null": None,
+                "integer": 1,
+                "float": 2.0,
+                "list": [1, 2.0, True, False],
+            },
+        }
```

Comment:
I think we're testing too much here. We don't need to test all the types (we can assume the `json` module works), just one or two. In addition, we're not testing string here, which is one of the more important types.

Perhaps `configs = {"n": 42, "s": "string", "d": {"nested": True}}`

Reply:
Yeah, that's a bit overkilled... Updated.